core/loadOptions.js

```diff
@@ -76,7 +76,7 @@ var legacyOptionsChecker = function(options, fileName){
 module.exports = function(basePath, _silent){
     silent = _silent;
 
-    var pathToApp = basePath || path.dirname(require.main.filename);
+    var pathToApp = basePath || global.pathToApp;
```

Comment:
You are right, it's just the rest of the codebase also needs to be improved in that sense :) But as I said, there's no better alternative, except passing the properties directly, as it's also done here in the main execution scenario.

But first let's discuss other SourceJS calling scenarios, since we may find a better option (following this discussion https://github.com/sourcejs/Source/issues/240#issuecomment-266748670).